src/AppInstallerCLICore/ContextOrchestrator.cpp

```diff
@@ -104,56 +109,56 @@ namespace AppInstaller::CLI::Execution
+            RemoveItemInState(*item, OrchestratorQueueItemState::Running);
 
             item = GetNextItem();
         }
     }
 
-    void ContextOrchestrator::RemoveItemInState(const std::shared_ptr<OrchestratorQueueItem>& item, OrchestratorQueueItemState state)
+    void ContextOrchestrator::RemoveItemInState(const OrchestratorQueueItem& item, OrchestratorQueueItemState state)
     {
         std::lock_guard<std::mutex> lock{ m_queueLock };
 
         // Look for the item. It's ok if the item is not found since multiple listeners may try to remove the same item.
-        auto itr = std::find(m_queueItems.begin(), m_queueItems.end(), item);
+        //auto itr = std::find(m_queueItems.begin(), m_queueItems.end(), item);
+        auto itr = FindIteratorById(item.GetId());
         if (itr != m_queueItems.end() && (*itr)->GetState() == state)
         {
             m_queueItems.erase(itr);
 
-            const auto& manifest = item->GetContext()->Get<Execution::Data::Manifest>();
+            const auto& manifest = item.GetContext().Get<Execution::Data::Manifest>();
             m_installingWriteableSource->RemovePackageVersion(manifest, std::filesystem::path{ manifest.Id + '.' + manifest.Version });
 
-            item->GetCompletedEvent().SetEvent();
+            item.GetCompletedEvent().SetEvent();
         }
     }
-    void ContextOrchestrator::CancelQueueItem(const std::shared_ptr<OrchestratorQueueItem>& item)
+    void ContextOrchestrator::CancelQueueItem(const OrchestratorQueueItem& item)
     {
         // Always cancel the item, even if it isn't running yet, to get the terminationHR set correctly.
-        item->GetContext()->Cancel(false, true);
+        item.GetContext().Cancel(false, true);
 
         RemoveItemInState(item, OrchestratorQueueItemState::Queued);
     }
 
-    std::shared_ptr<OrchestratorQueueItem> ContextOrchestrator::GetQueueItem(const OrchestratorQueueItemId* queueItemId)
+    std::shared_ptr<OrchestratorQueueItem> ContextOrchestrator::GetQueueItem(const OrchestratorQueueItemId& queueItemId)
     {
         std::lock_guard<std::mutex> lock{ m_queueLock };
 
         return FindById(queueItemId);
     }
 
-    bool OrchestratorQueueItemId::IsSame(const OrchestratorQueueItemId* comparedId)
+    bool OrchestratorQueueItemId::IsSame(const OrchestratorQueueItemId& comparedId) const
     {
-        return ((GetPackageId() == comparedId->GetPackageId()) && 
-                (GetSourceId() == comparedId->GetSourceId()));
+        return ((GetPackageId() == comparedId.GetPackageId()) && 
+                (GetSourceId() == comparedId.GetSourceId()));
     }
 
-    std::shared_ptr<OrchestratorQueueItem> OrchestratorQueueItemFactory::CreateItemForInstall(std::wstring packageId, std::wstring sourceId, std::shared_ptr<COMContext> context)
+    std::unique_ptr<OrchestratorQueueItem> OrchestratorQueueItemFactory::CreateItemForInstall(std::wstring packageId, std::wstring sourceId, std::unique_ptr<COMContext> context)

```

Comment:
On further inspection, I don't see how this even compiles.  The result of this is directly passed to a a function that takes a shared_ptr...